clang/test/SemaSYCL/reqd-sub-group-size-device.cpp

```diff
@@ -48,6 +48,17 @@ void bar() {
   kernel<class kernel_name5>([]() [[cl::intel_reqd_sub_group_size(2)]] { });
 }
 
+#ifdef TRIGGER_ERROR
+// expected-note@+1 {{conflicting attribute is here}}
+[[cl::intel_reqd_sub_group_size(2)]] void sg_size2() {}
+
+// expected-note@+2 {{conflicting attribute is here}}
+// expected-error@+1 {{conflicting attributes applied to a SYCL kernel}}
```

Comment:
This diagnostic doesn't seem to be correct, because function it is pointing to is not a SYCL kernel

Reply:
This diagnostic seems strange as well, why does it think that sg_size4 and sg_size2 are the same kernel?  

Should this conflict be diagnosed on a call expression instead?

Reply:
> This diagnostic doesn't seem to be correct, because function it is pointing to is not a SYCL kernel

OK

Reply:
> This diagnostic seems strange as well, why does it think that sg_size4 and sg_size2 are the same kernel?

Why do you think "it think that sg_size4 and sg_size2 are the same kernel" - please elaborate.

> Should this conflict be diagnosed on a call expression instead?

This is a question to existing design. I'm not changing anything in the diagnostics mechanism. I'm fixing existing bug where SYCL_EXTERNAL functions were missing in that diagnostics - I add them.

Reply:
From the error message, it says "conflicting attributes applied to a SYCL kernel".  From the notes, it specifies attributes on two separate declarations.

So by reading the error + notes, it implies that there are two conflicting attributes on a single kernel, yet they are on different functions.

Reply:
This error message existed before - I did not invent it, I just used it for SYCL_EXTERNAL function. Do you suggest to fix prior problems?

Reply:
We do this already?  We have other conditions where we diagnose an error on a kernel about conflicting attributes, and then point to an attribute on another declaration?

Reply:
Yep. SYCL attributes are propagated from called device function up to the kernel (AFAIK it is spec defined behavior which I find pretty magic), sometimes there are several functions called from kernel and their attribute values can conflict. Diagnosing happens here https://github.com/intel/llvm/blob/12d14e8194c8a36f70e4a7800d6f66388dcd0b65/clang/lib/Sema/SemaSYCL.cpp#L1484

Reply:
Ah, well, the way we report that is awful.  But not needing to be fixed in this patch.  Thanks @Fznamznon.  The error-message should likely be changed to be more clear about this if anyone ends up getting to it.